core/src/main/java/org/stellar/anchor/util/SepLanguageHelper.java

```diff
@@ -0,0 +1,105 @@
+package org.stellar.anchor.util;
+
+import static org.stellar.anchor.util.Log.errorEx;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import org.stellar.anchor.api.exception.SepValidationException;
+import org.stellar.anchor.config.AppConfig;
+
+public class SepLanguageHelper {
+  static Map<String, Iso3316Language> fallbackLangs = null;
+
+  public static String validateLanguage(AppConfig appConfig, String lang) {
+    if (fallbackLangs == null) {
+      fallbackLangs = prepareFallbackLanguages(appConfig);
+    }
+
+    if (lang != null) {
+      List<String> languages = appConfig.getLanguages();
+      if (languages != null && languages.size() > 0) {
+        if (languages.stream().noneMatch(l -> l.equalsIgnoreCase(lang))) {
+          return getFallbackLanguage(lang);
+        }
+      }
+      return lang;
+    } else {
+      return getFallbackLanguage(lang);
+    }
+  }
+
+  public static void reset() {
+    fallbackLangs = null;
+  }
+
+  static Map<String, Iso3316Language> prepareFallbackLanguages(AppConfig appConfig) {
+    Map<String, Iso3316Language> fallbacks = new HashMap<>();
+    appConfig
+        .getLanguages()
+        .forEach(
+            lang -> {
+              try {
+                Iso3316Language isoLang = Iso3316Language.of(lang);
+                fallbacks.putIfAbsent(isoLang.getLangKey(), isoLang);
+              } catch (SepValidationException ex) {
+                errorEx(ex);
+              }
+            });
+    return fallbacks;
+  }
+
+  static final String LANGUAGE_ONLY_DEFAULT = "en";
+  static final String LANGUAGE_DEFAULT = "en-US";
+
+  static String getFallbackLanguage(String lang) {
+    try {
+      Iso3316Language language = Iso3316Language.of(lang);
+      Iso3316Language fallback = fallbackLangs.get(language.getLangKey());
```

Comment:
💡  Here you're comparing both the language and locale. Maybe we could fallback to the language only. 

For instance, `pt` and `pt_br` would be considered different in your logic, while in practice they're about the same thing.

Maybe a change for another PR though.

Reply:
There was a discussion in slack between Cassio, Niranjan and me. Here is the test cases that may better illustrate the strategy. We may of course make changes in the future PR to make strategy change or even make them configurable. 

https://github.com/stellar/java-stellar-anchor-sdk/blob/c5a3f2b398c30fb998216508f655fd6eef54a26d/core/src/test/kotlin/org/stellar/anchor/util/SepLanguageHelperTest.kt#L37-L58